pkgs/development/libraries/criterion/default.nix

```diff
@@ -0,0 +1,43 @@
+{ stdenv, fetchFromGitHub, cmake, pkg-config, boxfort, libcsptr, dyncall, nanomsg }:
+
+stdenv.mkDerivation rec {
+  pname = "criterion";
+  version = "2.3.3";
+
+  src = fetchFromGitHub
+    { owner = "Snaipe";
+      repo = "Criterion";
+      rev = "v${version}";
+      fetchSubmodules = true;
+      sha256 = "0y1ay8is54k3y82vagdy0jsa3nfkczpvnqfcjm5n9iarayaxaq8p";
+    };
+
+  buildInputs =
+    [ boxfort
+      libcsptr
+      dyncall
+      nanomsg
+    ];
+
+  nativeBuildInputs =
+    [ cmake
+      pkg-config
+    ];
+
+  # CMake needs a little help to find BoxFort and pkg-config
+  preConfigure = ''
+    mkdir -p .third-party
+    cp -r ${boxfort} .third-party/boxfort
+    cp -r ${pkg-config} .third-party/pkg-config
```

Comment:
does `ln -s` work here? would be less expensive

Reply:
I saw that being used in another pull request discussion, will test with ln